dependencies.yaml

```diff
@@ -184,6 +184,7 @@ files:
     includes:
       - test_python_common
 channels:
+  - numba/label/ci
```

Comment:
It looks like this only affects the environment files (so offline testing). We may need a different approach if we want to test on CI

Reply:
Can you explain what you mean by offline testing? I see the following in the `conda-python-cudf-tests` CI jobs with the latest commit:

```
  + numba                                   0.59.0dev0  np1.21py3.10hfbc5805_g7c1ea796a_596  numba/label/ci          4MB
```

Reply:
Yeah it seems this is only used in environment files that are AFAIK not used in CI

Reply:
`dependencies.yaml` is used by CI:

- https://github.com/rapidsai/cudf/blob/cee642916cfc3b8df73e819bf3bc50f1b9fc684f/ci/test_cpp_common.sh#L12-L15

You can see the environment it generates for CI tests in the logs below:

- https://github.com/rapidsai/cudf/actions/runs/7211137353/job/19646850155?pr=14616#step:7:27

Reply:
It's not used in [the build though]( https://github.com/rapidsai/cudf/actions/runs/7211137353/job/19646852176#step:7:498 )

Also it appears [the tests don't use it either]( https://github.com/rapidsai/cudf/actions/runs/7211137353/job/19647545108#step:7:949 )

Reply:
@ajschmidt8 could we enforce numba 0.59 for testing by pinning the dependency [here](https://github.com/rapidsai/cudf/blob/branch-24.02/dependencies.yaml#L596C3-L596C19) perhaps? Maybe along with adding the extra channel back somewhere?

Reply:
Maybe we could fold these into one command

https://github.com/rapidsai/cudf/blob/cee642916cfc3b8df73e819bf3bc50f1b9fc684f/ci/test_python_common.sh#L19

https://github.com/rapidsai/cudf/blob/cee642916cfc3b8df73e819bf3bc50f1b9fc684f/ci/test_python_common.sh#L37-L40

Reply:
Tried installing it as a step at the end - will see what happens.

Reply:
With https://github.com/rapidsai/cudf/pull/14616/commits/5ffac4d6ab66b12c0ff3a24252b536ae23b4a78a and https://github.com/rapidsai/cudf/pull/14616/commits/8e67d620fa8e0748482e74b5f48c5c0eb77bf671 I think I managed to get it to install numba 0.59 just before the cuDF tests run. @jakirkham let me know if this seems like this covers us for now.

Reply:
Thanks Brandon! 🙏

Looking at [the Conda cuDF Python CI test]( https://github.com/rapidsai/cudf/actions/runs/7225028551/job/19688893979?pr=14616#step:7:944 ), it seems to oddly be getting an older version from `conda-forge`

```
  - numba                 0.58.1  py310h7dc5dd1_0                      conda-forge             Cached
  + numba                 0.57.1  py310h0f6aa51_0                      conda-forge                4MB
```

Reply:
Ah sorry was looking at the latest commit. Should I be looking at the checks on the commit right before?

If I do, I see the following in the [Conda cuDF Python CI test]( https://github.com/rapidsai/cudf/actions/runs/7223085574/job/19682887349#step:7:1079 ):

```
  Package        Version  Build                                Channel              Size
──────────────────────────────────────────────────────────────────────────────────────────
  Install:
──────────────────────────────────────────────────────────────────────────────────────────

  + zlib          1.2.13  hd590300_5                           conda-forge          93kB
  + numba     0.59.0dev0  np1.21py3.10ha54aa7d_gfaaf5f891_609  numba/label/ci        4MB

  Upgrade:
──────────────────────────────────────────────────────────────────────────────────────────

  - llvmlite      0.40.1  py310h1b8f574_0                      conda-forge        Cached
  + llvmlite  0.42.0dev0  py310he1b5a44_74                     numba/label/ci       35MB
```

This looks good to me

Thanks again for working on this Brandon! 🙏